src/installer/managed/Microsoft.NET.HostModel/AppHost/HostWriter.cs

```diff
@@ -103,6 +98,41 @@ void SetLastWriteTime()
 
             try
             {
+                RetryUtil.RetryOnIOError(() =>
+                {
+                    MemoryMappedFile memoryMappedFile = null;
+                    MemoryMappedViewAccessor memoryMappedViewAccessor = null;
+                    try
+                    {
+                        // Open the source host file.
+                        memoryMappedFile = MemoryMappedFile.CreateFromFile(appHostSourceFilePath, FileMode.Open, null, 0, MemoryMappedFileAccess.CopyOnWrite);
+                        memoryMappedViewAccessor = memoryMappedFile.CreateViewAccessor(0, 0, MemoryMappedFileAccess.CopyOnWrite);
+
+                        // Transform the host file in-memory.
+                        RewriteAppHost(memoryMappedViewAccessor);
+
+                        // Save the transformed host.
+                        BinaryUtils.SaveFile(memoryMappedViewAccessor, appHostDestinationFilePath);
+                    }
+                    finally
+                    {
+                        if (memoryMappedViewAccessor != null)
```

Comment:
can either of these actually be null here?

Reply:
Yes.  As an example, if `MemoryMappedFile.CreateFromFile` throws, then `memoryMappedViewAccessor` would be `null`.

Reply:
Oop..